pkg/operator/ceph/cluster/cleanup.go

```diff
@@ -182,3 +184,12 @@ func (c *ClusterController) getCephHosts(namespace string) ([]string, error) {
 
 	return hostNameList, nil
 }
+
+func (c *ClusterController) getMonSecret(namespace string) (string, error) {
```

Comment:
Don't you have `clusterInfo` from `clusterMap`? Also you know the `cluster` already so I don't think this is necessary. or just in case you could check with `IsInitialized` and then run this.

Reply:
Can't seem to get the monitor secret from ClusterMap (using both the `newCluster` and `existing` one in the `onUpdate` method). Running into `invalid memory address or nil pointer dereference` error.

Reply:
Sounds good.